security/advancedtls/examples/credential_reloading_from_files/client/main.go

```diff
@@ -99,55 +90,36 @@ func main() {
-	ctx, cancel := context.WithTimeout(context.Background(), defaultConnTimeout)
-	defer cancel()
-	oldConn, err := grpc.DialContext(ctx, address, grpc.WithTransportCredentials(clientTLSCreds))
-	if err != nil {
-		log.Fatalf("grpc.DialContext to %s failed: %v", address, err)
-	}
-	oldClient := pb.NewGreeterClient(oldConn)
-	_, err = oldClient.SayHello(ctx, &pb.HelloRequest{Name: "gRPC"}, grpc.WaitForReady(true))
-	if err != nil {
-		log.Fatalf("oldClient.SayHello failed: %v", err)
-	}
-	defer oldConn.Close()
-
-	// Change the contents of the credential files.
-	// Cert "another_client_cert_1.pem" is also trusted by "server_cert_1.pem".
-	if err := copyFileContents(testdata.Path("another_client_cert_1.pem"), certTmpFile.Name()); err != nil {
-		log.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("another_client_cert_1.pem"), certTmpFile.Name(), err)
-	}
-	if err := copyFileContents(testdata.Path("another_client_key_1.pem"), keyTmpFile.Name()); err != nil {
-		log.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("another_client_key_1.pem"), keyTmpFile.Name(), err)
-	}
+	// Send the requests every 0.5s. The credential is expected to be changed in
+	// the bash script.
+	ticker := time.NewTicker(500 * time.Millisecond)
+	done := make(chan bool)
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			case <-ticker.C:
+				// Make a connection using the credentials.
+				ctx, cancel := context.WithTimeout(context.Background(), defaultConnTimeout)
+				conn, err := grpc.DialContext(ctx, address, grpc.WithTransportCredentials(clientTLSCreds))
+				if err != nil {
+					log.Fatalf("grpc.DialContext to %s failed: %v", address, err)
+				}
+				client := pb.NewGreeterClient(conn)
+				_, err = client.SayHello(ctx, &pb.HelloRequest{Name: "gRPC"}, grpc.WaitForReady(true))
+				if err != nil {
+					log.Fatalf("client.SayHello failed: %v", err)
+				}
+				cancel()
+				conn.Close()
+			}
+		}
+	}()
 
-	// Wait for 1 second for the provider to load the new contents.
-	time.Sleep(time.Second)
-
-	// Make a connection using the new credential contents, while the credential
-	// struct remains the same.
-	// The common name of the old and new certificate will be logged on the
-	// server side.
-	newConn, err := grpc.DialContext(ctx, address, grpc.WithTransportCredentials(clientTLSCreds))
-	if err != nil {
-		log.Fatalf("grpc.DialContext to %s failed: %v", address, err)
-	}
-	newClient := pb.NewGreeterClient(newConn)
-	_, err = newClient.SayHello(ctx, &pb.HelloRequest{Name: "gRPC"}, grpc.WaitForReady(true))
-	if err != nil {
-		log.Fatalf("newClient.SayHello failed: %v", err)
-	}
-}
-
-func copyFileContents(sourceFile, destinationFile string) error {
-	input, err := ioutil.ReadFile(sourceFile)
-	if err != nil {
-		return err
-	}
-	err = ioutil.WriteFile(destinationFile, input, 0644)
-	if err != nil {
-		return err
-	}
-	return nil
+	// Stop the ticker after 3s.
+	time.Sleep(3000 * time.Millisecond)
```

Comment:
Is this long enough for the test to be successful?

Maybe it's better to run indefinitely, and have the script kill it?